qa/rpc-tests/test_framework/util.py

```diff
@@ -78,8 +90,16 @@ def initialize_chain(test_dir):
     bitcoind and bitcoin-cli must be in search path.
     """
 
-    if not os.path.isdir(os.path.join("cache", "node0")):
-        devnull = open("/dev/null", "w+")
+    if ( not os.path.isdir(os.path.join("cache", "node0"))
+             or not os.path.isdir(os.path.join("cache", "node1")) 
+             or not os.path.isdir(os.path.join("cache", "node2")) 
+             or not os.path.isdir(os.path.join("cache", "node3")) ):
+
+        #find and delete old cache directories if any exist
+        for i in range(4):
+	    if os.path.isdir(os.path.join("cache", "node"+str(i))): shutil.rmtree(os.path.join("cache", "node"+str(i)))
```

Comment:
The purpose is to delete any old directories "if they exist" before 
creating the cache files anew. For instance, when I found this issue I 
was just beginning to debug the code and eventually
found that for some reason only the cache for node0  had been created.  
The old code would assume that if the cache for node0 was found then 
everything was OK
however the scripts woul fail later on.   With the nested loop if a 
cache dir is found then it will be deleted so that in the next step all 
four can be recreated
to ensure that the cache is complete.

On 16/08/2015 5:24 AM, MarcoFalke wrote:
>
> In qa/rpc-tests/test_framework/util.py 
> <https://github.com/bitcoin/bitcoin/pull/6548#discussion_r37145281>:
>
> > @@ -78,8 +90,16 @@ def initialize_chain(test_dir):
> >      bitcoind and bitcoin-cli must be in search path.
> >      """
> >
> > -    if not os.path.isdir(os.path.join("cache", "node0")):
> > -        devnull = open("/dev/null", "w+")
> > +    if ( not os.path.isdir(os.path.join("cache", "node0"))
> > +             or not os.path.isdir(os.path.join("cache", "node1"))
> > +             or not os.path.isdir(os.path.join("cache", "node2"))
> > +             or not os.path.isdir(os.path.join("cache", "node3")) ):
> > +
> > +        #find and delete old cache directories if any exist
> > +        for i in range(4):
> > +	    if os.path.isdir(os.path.join("cache", "node"+str(i))): shutil.rmtree(os.path.join("cache", "node"+str(i)))
>
> What is the reason to nest the for loop inside the |if|?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/bitcoin/bitcoin/pull/6548/files#r37145281>.
>